config/application.rb

```diff
@@ -60,10 +62,20 @@ class Application < Rails::Application
     # Allow deserialization of Time objects:
     config.active_record.yaml_column_permitted_classes = [Time]
 
+    # # Make belongs_to optional: true
+    # config.active_record.belongs_to_required_by_default = false
+
     # Add custom delivery method for emails
     ActionMailer::Base.add_delivery_method(:notify_mail, NotifyMail)
 
-    # For Rails 7: https://guides.rubyonrails.org/active_record_multiple_databases.html#migrate-to-the-new-connection-handling
-    config.active_record.legacy_connection_handling = false
+    # Rails secrets is deprecated in favour of Rails credentials, but that
+    # doesn't work for us so we want to mimic Rails secrets functionality:
+    def self.secrets
+      @secrets ||= begin
+        secrets = ActiveSupport::OrderedOptions.new
+        files = config.paths["config/secrets"].existent
+        secrets.merge! Rails::Secrets.parse(files, env: Rails.env)
```

Comment:
that's clever, but are we sure that Rails::Secrets.parse and the entire Rails::Secrets class won't be deprecated going forward?

Reply:
Oops good point! I'll change this.